apps/lib/apps.c

```diff
@@ -2720,3 +2721,30 @@ void app_params_free(OSSL_PARAM *params)
         OPENSSL_free(params);
     }
 }
+
+int app_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY **pkey, const char *alg, int bits)
+{
+    if (alg != NULL) {
+        if (bits > 0)
+            BIO_printf(bio_err, "Generating a %s key, with %d bits\n",
+                       alg, bits);
+        else
+            BIO_printf(bio_err, "Generating a %s key\n", alg);
+    }
+    if (!RAND_status())
+        BIO_printf(bio_err, "Warning: generating random key material may take long time if the system has a poor entropy source\n");
+    if (EVP_PKEY_keygen(ctx, pkey) > 0)
+        return 1;
```

Comment:
Will the keygen callback be pushing errors on the stack that we need to print?  There are lots of code paths that call these helpers now but don't seem to have an `ERR_print_errors()` anymore.

Reply:
See my remark on `ERR_print_errors()` above.